torch/distributed/fsdp/sharded_grad_scaler.py

```diff
@@ -0,0 +1,329 @@
+from collections import abc, defaultdict
+import logging
+from typing import Dict, List, Optional, Union
+
+import torch
+from torch.cuda import FloatTensor  # type: ignore
+from torch.cuda.amp.grad_scaler import GradScaler, OptState, _MultiDeviceReplicator
+from torch.distributed.distributed_c10d import ProcessGroup
+import torch.distributed as dist
+from torch.optim.sgd import SGD
+
+
+def _refresh_per_optimizer_state():
+    return {"stage": OptState.READY, "found_inf_per_device": {}}
+
+
+def _is_supported_device(tensor: torch.Tensor):
+    return tensor.is_cuda or tensor.device.type in ("xla", "cpu")
```

Comment:
FSDP doesn't support XLA at the moment so we can possibly omit, though it might just be worth keeping for when we eventually do support XLA, I'll let you make the call